### DISCUSSION TITLE
"Connector not found" on first render

### DISCUSSION THREAD

**Discussion by MarioCodetria** [wevm/wagmi#1772] [Q&A] [answered]
I'm working on a React JS app, using Wagmi hooks. However, for this specific case, I'm using @wagmi/core signMessage, because I need to sign a message outside a component, so I can't use hooks here. I need to sign a message right before doing an API call. This call runs when my page is mounting. If I try to sign the message at that time, I get this error: "_ConnectorNotFoundError: Connector not found_". If I run the same code a couple seconds later, it works fine. I suppose this happens because something is not initialized yet. Is there a way to know _when_ it is ready to sign a message?

**Comment by tmm** [comment, marked as answer]
Use the `wagmi/actions` entrypoint instead of importing from `@wagmi/core`:

```ts 
import { signMessage } from 'wagmi/actions'
```

Then check whether or not a wallet is connected using `getAccount` or wrap in a try-catch block.

**Comment by MarioCodetria** [comment]
Yeah, I ended up using a catch block to retry a couple more times after a short delay if it fails. Thanks.

**Comment by di-sukharev** [comment]
but i want to call it on the server.. and importing from 'wagmi/actions' errors that is can not be called on server
